detect/detect.go

```diff
@@ -272,6 +273,9 @@ func (d *Detector) detectRule(fragment Fragment, rule config.Rule) []report.Find
 			loc.endLineIndex = matchIndex[1]
 		}
 
+		// Calculate the SHA256 hash sum of secret
+		secretSHA := fmt.Sprintf("%x", sha256.Sum256([]byte(secret)))
```

Comment:
How do you feel about calling `strings.TrimSpaces` before hashing the secret? I've seen a few instances where the same secret is captured with various amounts of whitespace, and doing so would ensure they all produce the same hash.

Reply:
Hi @rgmz . 
Currently, the variable Secret is trimmed by new line occurrences ('\n') and the hash in SecretSHA uses the same value from the Secret variable for consistency. If you are about to add additional trim of spaces (" ") to the Secret variable - I suggest doing it as a separate Pull request with an example of the issue.